### DISCUSSION TITLE
Move the 'Add ctrl' button to the toolbar?

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1171] [Team Posts]
In view of @kybos 's new Arranger scrollbar, I have an equally brilliant idea:

Let's move the Add Ctrl button to the toolbar!
Oh wait... That's @kybos ' idea, already stated ;-)
Gee... did I argue against that? Seemed right at the time.

I'm thinking now that it should be done, so that we might add a scrollbar
 in the editors as well. Maybe I'll try today...

**Comment by terminator356** [comment]
Space for rent!

- Midi controller graphs: Moved 'Add ctrl' button from bottom area to menu and toolbar as well.

Don't worry:
1) The popup menu is shared between the main menu and the toolbar which I think is legal
2) It retains its 'stay open' goodness
3) It is still updated dynamically according to the current chosen instrument -
 cleared and repopulated each time opened, via the use of QMenu::aboutToShow().

Give 'er a whirl.

@kybos Well, at least that frees up one more item on the bottom.
Steps towards bottom area independence from scrollbar height?
I realized just now that adding this new scrollbar of yours to DrumEditor might be tricky:
We have controller lanes sitting between the drum list and the bottom area.
Worse, the controller 'panels' are not scrollable things so it might seem a little weird
 having a scrollbar at the bottom ;-( It really could use one there.

**Comment by kybos** [comment]
> Midi controller graphs: Moved 'Add ctrl' button from bottom area to menu and toolbar as well.

Thanks! Another important step towards a more standard-conform UI ;-).

Is there any technical reason not to use a normal dropdown button in the toolbar? It looks quite inconsistent, especially as there is the _Play sound_ dropdown button close to it (the MIDI panic is another one):
![image](https://user-images.githubusercontent.com/10009541/85196558-42694a80-b2db-11ea-8f71-2dd2de1724de.png)

> I realized just now that adding this new scrollbar of yours to DrumEditor might be tricky:
> We have controller lanes sitting between the drum list and the bottom area.
> Worse, the controller 'panels' are not scrollable things so it might seem a little weird
> having a scrollbar at the bottom ;-( It really could use one there.

Yes, I had realized that too. I am not sure how bad this would be. Certainly not quite standard, but it would add symmetry and make good use of the available space. You know I am very picky about the UI, but I think I could live with this one ;-).
What would be your votes guys? Pro or contra the scrollbar?

**Comment by terminator356** [comment]
The other two buttons have two states, so the arrow must be to the right.
This button doesn't, so the opportunity to save a few pixels arose.
I suppose we could change it, super easy.
[ Edit ] The panic doesn't have two states but is clickable. Controller button is not.

**Comment by kybos** [comment]
> The other two buttons have two states, so the arrow must be to the right.
> This button doesn't, so the opportunity to save a few pixels arose.
> I suppose we could change it, super easy.
> [ Edit ] The panic doesn't have two states but is clickable. Controller button is not.

Yes I know what you mean, and technically you are absolutely right, the arrow part would be kind of redundant here. I'm just a little afraid that the average user won't do this kind of deep analysis ;-), he will just find the UI look inconsistent.
I think your icon with the arrow would be perfect for the "Select controller" button in the controller area, as there is really no room for a genuine drop-down button. In the toolbar perhaps your original cogwheel with the plus sign would be clearer.
